src/App.jsx

```diff
@@ -1,24 +1,205 @@
+flex-direction: column;
+align-items: center;
+justify-content: center;
+height: 100vh; /* 100% of the viewport height */
+text-align: center; /* Ensures that text inside the container is centered */
+`;
+
+
+const TitleContainer = styled.div`
+  position: center; /* Set to relative to act as a reference for the absolute positioned Lottie container */
+  display: inline-block; /* Ensures the container doesn't take full width */
+`;
+
+const StyledH1 = styled.h1`
+  color: #01056F;
+  font-size: 2.5rem;
+  text-align: center;
+  position: relative; /* Needed for z-index stacking context */
+  z-index: 2; /* Ensures the text is above the Lottie animation */
+`;
+
+const StyledH2 = styled.h2`
+  color: #01056F;
+  font-size: 2rem;
+`;
+
+const LottieContainer = styled.div`
+  margin-bottom: -30px; // Or however much space you want between the animation and the title
+  width: 250px;
+  height: 250px;
+  border: outset;
+  border-width:1px;
+ color: #01056F;
+`;
+
+const QRCodeImage = styled.img`
+  display: block;
+  margin: auto; /* This will center the image horizontally */
+  margin: 20px;
+`;
+
+
+const ModalBackdrop = styled.div`
+  position: fixed;
+  z-index: 1000;
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  background-color: rgba(0, 0, 0, 0.5);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const ModalContent = styled.div`
+  background: white;
+  padding: 20px;
+  border-radius: 5px;
+  text-align: center;
+`;
+
+const ErrorMessage = styled.div`
+  color: red;
+  margin-top: 8px;
+  font-size: 14px;
+`;
+
+const StyledButton = styled.button`
+  background-color: #01056F;
+  color: white;
+  padding: 10px 20px;
+  text-align: center;
+  text-decoration: none;
+  display: inline-block;
+  font-size: 18px;
+  font-weight: bold; /* Makes the text bold */
+  margin: 4px 2px;
+  transition-duration: 0.4s;
+  cursor: pointer;
+  border-radius: 30px;
+`;
+
+const App = () => {
+  const { url, setUrl, qrCode, showInput, generateQRCode, resetQRCode } = useQRCodeGenerator();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [fileName, setFileName] = useState('');
+
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const defaultOptions = {
+    loop: true,
+    autoplay: true,
+    animationData: animationData,
+    rendererSettings: {
+      preserveAspectRatio: 'xMidYMid slice'
+    }
+  };
+
+  const handleGenerate = (e) => {
+    e.preventDefault();
+    const validUrlRegex = /^(https?:\/\/(www\.)?|www\.)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/i;
+
+
+  
+    if (validUrlRegex.test(url)) {
+      generateQRCode(url);
+      setErrorMessage(''); // Clear any previous error messages
+    } else {
+      setErrorMessage("Please enter a valid url starting with http, https or www.");
+    }
+  };
+  
+
+  const handleDownload = () => {
+    if (!fileName) {
+      setIsModalOpen(true);
+      return;
+    }
+
+    const link = document.createElement('a');
```

Comment:
I don't really understand why this one was repeated here as you dealt with it in your special hook. I personally think that it would be neater to have this part there instead of in App.